### PR TITLE
Slow down angle arm when it nears the extents of its range

### PR DESCRIPTION
**Pull request by mmcadamsms**

### CODE REVIEW

**Review by josieammer — APPROVED** [review]
LGTM!  Want to test on the robot before merging?

File: smokey/src/main/java/frc/robot/subsystems/ArmSubsystem.java [hunk 1]
```diff
@@ -373,6 +384,22 @@ else if(isVerticalArmAtOrAboveHighStop && this.requestedVerticalArmExtension >=
...
+        else if(currentVerticalExtensionInMeters > lengthVerticalArmExtensionVeryCloseToPucksMeters) {
+          double frogSpellExtensionDistance = 
+            (currentVerticalExtensionInMeters + this.requestedVerticalArmExtension) / 2;
+          verticalPidController.setReference(
+            ArmSubsystem.convertVerticalArmExtensionFromMetersToTicks(frogSpellExtensionDistance),
+            ControlType.kSmartMotion);
+        }
+        // we are nearing bottom stop zone sometimes too fast
+        else if(currentVerticalExtensionInMeters < lengthVerticalArmExtensionVeryCloseToStopMeters) {
+          double frogSpellExtensionDistance =
```
Comment (mmcadamsms) [inline smokey/src/main/java/frc/robot/subsystems/ArmSubsystem.java:397]: This can be combined with the above block.

**Review by mmcadamsms — COMMENTED** [review]
combine?

File: smokey/src/main/java/frc/robot/subsystems/ArmSubsystem.java (hunk 1 above)
Comment (josieammer) [reply smokey/src/main/java/frc/robot/subsystems/ArmSubsystem.java:397]: I think we should leave it as is.

**Review by josieammer — APPROVED** [review]
LGTM!

**Pull request merged**